src/module_2/EDA_2.md

```diff
@@ -0,0 +1,1265 @@
+      <td>NaN</td>
+      <td>3.643302e+12</td>
+      <td>5.029635e+12</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>2.100000e+01</td>
+      <td>1.000000e+00</td>
+      <td>1.000000e+00</td>
+      <td>1.000000e+00</td>
+      <td>...</td>
+      <td>3.000000e+00</td>
+      <td>1.000000e+00</td>
+      <td>6.000000e+00</td>
+      <td>5.000000e+00</td>
+      <td>1.480000e+02</td>
+      <td>8.400000e+01</td>
+      <td>5.868986e+01</td>
+      <td>1.480000e+02</td>
+      <td>3.950000e+01</td>
+      <td>3.564191e+01</td>
+    </tr>
+  </tbody>
+</table>
+<p>11 rows × 27 columns</p>
+</div>
+
+
+
+
+```python
+# Contar los valores nulos por columna
+df.isnull().sum()
+
+```
+
+
+
+
+    variant_id                          0
+    product_type                        0
+    order_id                            0
+    user_id                             0
+    created_at                          0
+    order_date                          0
+    user_order_seq                      0
+    outcome                             0
+    ordered_before                      0
+    abandoned_before                    0
+    active_snoozed                      0
+    set_as_regular                      0
+    normalised_price                    0
+    discount_pct                        0
+    vendor                              0
+    global_popularity                   0
+    count_adults                        0
+    count_children                      0
+    count_babies                        0
+    count_pets                          0
+    people_ex_baby                      0
+    days_since_purchase_variant_id      0
+    avg_days_to_buy_variant_id          0
+    std_days_to_buy_variant_id          0
+    days_since_purchase_product_type    0
+    avg_days_to_buy_product_type        0
+    std_days_to_buy_product_type        0
+    dtype: int64
+
+
+
+
+```python
+# Verificar duplicados
+duplicados = df.duplicated().sum()
+print(f"Número de filas duplicadas: {duplicados}")
+
+
+```
+
+    Número de filas duplicadas: 0
+
+
+
+```python
+# Si hay duplicados, podemos eliminarlos
+df = df.drop_duplicates()
+```
+
+
+```python
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+# Separar las variables categóricas y numéricas
+variables_categoricas = ['product_type', 'vendor']
+variables_numericas = [
+    'variant_id', 'order_id', 'user_id', 'user_order_seq', 'outcome', 'ordered_before', 'abandoned_before', 
+    'active_snoozed', 'set_as_regular', 'normalised_price', 'discount_pct', 'global_popularity', 
+    'count_adults', 'count_children', 'count_babies', 'count_pets', 'people_ex_baby',
+    'days_since_purchase_variant_id', 'avg_days_to_buy_variant_id', 'std_days_to_buy_variant_id',
+    'days_since_purchase_product_type', 'avg_days_to_buy_product_type', 'std_days_to_buy_product_type'
+]
+
+# Crear gráficos para las variables categóricas
+for var in variables_categoricas:
+    plt.figure(figsize=(50, 10))
```

Comment:
I recommend again using fig, ax = plt.subplots() and pass it to seaborn so then you have better control to the axes